ArduPlane/commands_logic.cpp

```diff
@@ -603,6 +662,39 @@ bool Plane::verify_nav_wp(const AP_Mission::Mission_Command& cmd)
     return false;
 }
 
+/*
+  update navigation for the navigating into wind waypoints. Return true when the
+  waypoint is complete. Just like verify_nav_wp(), but without the custom distance checking.
+ */
+bool Plane::verify_nav_into_wind(const AP_Mission::Mission_Command& cmd)
+{
+    steer_state.hold_course_cd = -1;
+
+    if (auto_state.no_crosstrack) {
+        nav_controller->update_waypoint(current_loc, next_WP_loc);
+    } else {
+        nav_controller->update_waypoint(prev_WP_loc, next_WP_loc);
+    }
+
+    float acceptance_distance = nav_controller->turn_distance(g.waypoint_radius, auto_state.next_turn_angle);
+    if (auto_state.wp_distance <= acceptance_distance) {
+        gcs_send_text_fmt(MAV_SEVERITY_INFO, "Reached waypoint #%i dist %um",
+                          (unsigned)mission.get_current_nav_cmd().index,
+                          (unsigned)get_distance(current_loc, next_WP_loc));
+        return true;
+	}
+
+    // have we flown past the waypoint?
+    if (location_passed_point(current_loc, prev_WP_loc, next_WP_loc)) {
```

Comment:
You're re-inventing the wheel here. What is different about this waypoint than a normal waypoint? Only difference is it's generated but the mission system doesn't know that.